app/controllers/admin_controller.rb

```diff
@@ -8,4 +8,19 @@ def index
   def show
     @invoice = Invoice.find(params[:id])
   end
+
+  def show_2 
+    @merchant = Merchant.find(params[:merchant_id])
```

Comment:
Can we just have one show and have both invoice and merchant, because if we are routing it to the show it should have access to both methods even if we dont use the one, just my thoughts with a quick glance. Ill follow up with you for real